apps/blockchain/test/blockchain/chain_test.exs

```diff
@@ -14,15 +14,15 @@ defmodule Blockchain.ChainTest do
     b =
       "some data"
       |> Block.generate_next_block()
-      |> ProofOfWork.compute()
+      |> proof_of_work().compute()
```

Comment:
I have question because you are 100% more involve than me. 

What would happen if we want to support something like PoW + PoT together, IF I am not mistaking Dash coin does that, it split it up into 50% 50% work.

Also, let say tomorrow I want to create ProofOfSharing (PoSH) how would that look like?

When I told you to move it into some module I was actually thinking on create some config with the module (not called `proof_of_work` because defeat the idea 😄 ) where we can setup either a list or just a module and you will call the `compute` function on it, which is why I told you about the protocol.

So that way you can have as many implementations of how you reward or do any computation of this kind.

Something like 
```
config :blockchain,
  some_better_name_key: [ProofOfWork, ProofOfStake, ProofOfShares]

# or

config :blockchain,
  some_better_name_key: [
    pow: ProofOfWork, # <- for the sack of the demo whatever key value pair
    pos: ProofOfStake,
    posh: ProofOfShares
  ]
```

the last example could be useful when you want to control that you HAVE to use specific module for X reason.  I would even stick with the array for now 😄  we need study what other coins do differently and more important where the crypto world is moving to

Reply:
I understand what's your point here. However for now I only know proof-of-work, I would have to dig more into other protocols that replace PoW to have a better idea on how this could be implemented and how we could go to what you suggest here. If that's ok with you, I'll move your really good comment to an issue and spend times on it later

Reply:
issue here: https://github.com/robinmonjo/blockchain/issues/8